File: Mercury/person.m
:- module person.

:- interface.
:- import_module io.
:- pred main(io::di, io::uo) is det.

:- implementation.
:- import_module int, string.

:- type person
    ---> person(
            name :: string,
            age :: int
        ).

main(!IO) :-
    P = person("John", 21),
    io.write_string(P^name, !IO),
    
    P = person(_, Age),
    io.write_int(Age, !IO),
    io.nl(!IO).

File: OCaml/person.ml
let person = object
  val mutable name = "?"
  val mutable age = 0

  method inc_age =
    age <- age + 1

  method setname newname =
    name <- newname
    
  method getname =
    name
end;;

person#setname "John";
print_endline person#getname;;
